### PR TITLE
Backports v1.0 - #2774, #2768, #2788, #2426, #2744, #1221, #2831, #2894

### PR DESCRIPTION
**Pull request by seriva**
Contains the following backports:

Azure related can be tested at the same time:
- ✔️ Backport #2784 for issue #2774
- ✔️ Backport #2813 for issue #2768

Opendistro related touching the same files and can be tested at the same time:
- ✔️ Backport #2823 for issue #2788 
- ✔️ Backport #2442 for issue #2426
- ✔️ Backport #2764 for issue #2744

Few line-change in kafka-exporter and postgres-exporter: 
- ✔️ Backport #2776 for issue #1221
- ✔️ Backport #2831

Others:
- ✔️ Backport #2912 for issue #2894

### CODE REVIEW

File: CHANGELOG-1.0.md [hunk 1]
```diff
@@ -17,6 +18,12 @@
 - [#1294](https://github.com/epiphany-platform/epiphany/issues/1294) - Implement proper merging of lists of dictionaries for epicli yaml docs
 - [#1370](https://github.com/epiphany-platform/epiphany/issues/1370) - Epicli does not correctly generate vars for Postgres
 - [#1296](https://github.com/epiphany-platform/epiphany/issues/1296) - Epicli does not interpret alternative yaml boolean values as true booleans
+- [#2774](https://github.com/epiphany-platform/epiphany/issues/2774) - Issue creating service principle on Azure
```
Comment (erzetpe) [inline CHANGELOG-1.0.md]: Should be principal.

**Review by erzetpe — CHANGES_REQUESTED** [review]

**Review by erzetpe — DISMISSED** [review]

**Comment by przemyslavic** [comment]
Please also add #2822

**Comment by przemyslavic** [comment]
@seriva
:x: OpenDistro installation is broken.

v1.0.1 works fine
I also excluded log4j patch and it's still broken so that's not the cause.




```
2022-01-20T14:48:55.9572168Z[38;21m14:48:55 INFO cli.engine.ansible.AnsibleCommand - TASK [opendistro_for_elasticsearch : Check if default admin user exists] *******
2022-01-20T14:48:56.6826136Z[31;21m14:48:56 ERROR cli.engine.ansible.AnsibleCommand - FAILED - RETRYING: Check if default admin user exists (60 retries left).
...
2022-01-20T14:50:20.8665820Z[;21m14:50:20 ERROR cli.engine.ansible.AnsibleCommand - FAILED - RETRYING: Check if default admin user exists (1 retries left).
2022-01-20T14:50:22.2985300Z[31;21m14:50:22 ERROR cli.engine.ansible.AnsibleCommand - fatal: [ci-v10azurcentflannel-logging-vm-1.azurcent]: FAILED! => {"attempts": 60, "changed": false, "content_length": "637", "content_type": "application/json; charset=UTF-8", "elapsed": 0, "json": {"error": {"reason": "Cannot invoke \"com.amazon.opendistroforelasticsearch.security.securityconf.ConfigModel.mapSecurityRoles(com.amazon.opendistroforelasticsearch.security.user.User, org.elasticsearch.common.transport.TransportAddress)\" because \"this.configModel\" is null", "root_cause": [{"reason": "Cannot invoke \"com.amazon.opendistroforelasticsearch.security.securityconf.ConfigModel.mapSecurityRoles(com.amazon.opendistroforelasticsearch.security.user.User, org.elasticsearch.common.transport.TransportAddress)\" because \"this.configModel\" is null", "type": "null_pointer_exception"}], "type": "null_pointer_exception"}, "status": 500}, "msg": "Status code was 500 and not [200, 404]: HTTP Error 500: Internal Server Error", "redirected": false, "status": 500, "url": "https://10.1.3.5:9200/_opendistro/_security/api/internalusers/admin"}
```
From what I remember you have experienced this problem before.

```
[2022-01-20T15:43:40,536][WARN ][stderr                   ] [ec2-1-1-1-1] java.util.ServiceConfigurationError: com.sun.tools.attach.spi.AttachProvider: Provider sun.tools.attach.AttachProviderImpl could not be instantiated
[2022-01-20T15:43:40,537][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:584)
[2022-01-20T15:43:40,537][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:806)
[2022-01-20T15:43:40,549][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.util.ServiceLoader$ProviderImpl.get(ServiceLoader.java:724)
[2022-01-20T15:43:40,550][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.util.ServiceLoader$3.next(ServiceLoader.java:1396)
[2022-01-20T15:43:40,550][WARN ][stderr                   ] [ec2-1-1-1-1]        at jdk.attach/com.sun.tools.attach.spi.AttachProvider.providers(AttachProvider.java:261)
[2022-01-20T15:43:40,550][WARN ][stderr                   ] [ec2-1-1-1-1]        at jdk.attach/com.sun.tools.attach.VirtualMachine.attach(VirtualMachine.java:200)
[2022-01-20T15:43:40,551][WARN ][stderr                   ] [ec2-1-1-1-1]        at com.amazon.opendistro.elasticsearch.performanceanalyzer.jvm.ThreadList.runAttachDump(ThreadList.java:144)
[2022-01-20T15:43:40,551][WARN ][stderr                   ] [ec2-1-1-1-1]        at com.amazon.opendistro.elasticsearch.performanceanalyzer.jvm.ThreadList.lambda$runThreadDump$6(ThreadList.java:246)
[2022-01-20T15:43:40,551][WARN ][stderr                   ] [ec2-1-1-1-1]        at com.amazon.opendistro.elasticsearch.performanceanalyzer.core.Util.lambda$invokePrivileged$1(Util.java:55)
[2022-01-20T15:43:40,552][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
[2022-01-20T15:43:40,552][WARN ][stderr                   ] [ec2-1-1-1-1]        at com.amazon.opendistro.elasticsearch.performanceanalyzer.core.Util.invokePrivileged(Util.java:51)
[2022-01-20T15:43:40,552][WARN ][stderr                   ] [ec2-1-1-1-1]        at com.amazon.opendistro.elasticsearch.performanceanalyzer.jvm.ThreadList.runThreadDump(ThreadList.java:246)
[2022-01-20T15:43:40,552][WARN ][stderr                   ] [ec2-1-1-1-1]        at com.amazon.opendistro.elasticsearch.performanceanalyzer.jvm.ThreadList.getNativeTidMap(ThreadList.java:110)
[2022-01-20T15:43:40,553][WARN ][stderr                   ] [ec2-1-1-1-1]        at com.amazon.opendistro.elasticsearch.performanceanalyzer.collectors.OSMetricsCollector.collectMetrics(OSMetricsCollector.java:57)
[2022-01-20T15:43:40,553][WARN ][stderr                   ] [ec2-1-1-1-1]        at com.amazon.opendistro.elasticsearch.performanceanalyzer.collectors.PerformanceAnalyzerMetricsCollector.lambda$run$0(PerformanceAnalyzerMetricsCollector.java:60)
[2022-01-20T15:43:40,553][WARN ][stderr                   ] [ec2-1-1-1-1]        at com.amazon.opendistro.elasticsearch.performanceanalyzer.core.Util.lambda$invokePrivileged$1(Util.java:55)
[2022-01-20T15:43:40,553][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
[2022-01-20T15:43:40,554][WARN ][stderr                   ] [ec2-1-1-1-1]        at com.amazon.opendistro.elasticsearch.performanceanalyzer.core.Util.invokePrivileged(Util.java:51)
[2022-01-20T15:43:40,554][WARN ][stderr                   ] [ec2-1-1-1-1]        at com.amazon.opendistro.elasticsearch.performanceanalyzer.collectors.PerformanceAnalyzerMetricsCollector.run(PerformanceAnalyzerMetricsCollector.java:60)
[2022-01-20T15:43:40,554][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
[2022-01-20T15:43:40,563][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
[2022-01-20T15:43:40,563][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.lang.Thread.run(Thread.java:832)
[2022-01-20T15:43:40,564][WARN ][stderr                   ] [ec2-1-1-1-1] Caused by: java.security.AccessControlException: access denied ("com.sun.tools.attach.AttachPermission" "createAttachProvider")
[2022-01-20T15:43:40,564][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
[2022-01-20T15:43:40,564][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.security.AccessController.checkPermission(AccessController.java:1036)
[2022-01-20T15:43:40,565][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:408)
[2022-01-20T15:43:40,565][WARN ][stderr                   ] [ec2-1-1-1-1]        at jdk.attach/com.sun.tools.attach.spi.AttachProvider.<init>(AttachProvider.java:94)
[2022-01-20T15:43:40,565][WARN ][stderr                   ] [ec2-1-1-1-1]        at jdk.attach/sun.tools.attach.HotSpotAttachProvider.<init>(HotSpotAttachProvider.java:47)
[2022-01-20T15:43:40,565][WARN ][stderr                   ] [ec2-1-1-1-1]        at jdk.attach/sun.tools.attach.AttachProviderImpl.<init>(AttachProviderImpl.java:38)
[2022-01-20T15:43:40,566][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2022-01-20T15:43:40,566][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:64)
[2022-01-20T15:43:40,566][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2022-01-20T15:43:40,566][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2022-01-20T15:43:40,567][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2022-01-20T15:43:40,567][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.util.ServiceLoader$ProviderImpl$2.run(ServiceLoader.java:790)
[2022-01-20T15:43:40,567][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.security.AccessController.doPrivileged(AccessController.java:691)
[2022-01-20T15:43:40,567][WARN ][stderr                   ] [ec2-1-1-1-1]        at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:795)
[2022-01-20T15:43:40,568][WARN ][stderr                   ] [ec2-1-1-1-1]        ... 20 more
[2022-01-20T15:43:42,090][WARN ][c.a.o.s.c.Salt           ] [ec2-1-1-1-1] If you plan to use field masking pls configure compliance salt e1ukloTsQlOgPquJ to be a random string of 16 chars length identical on all nodes
[2022-01-20T15:43:42,133][INFO ][c.a.o.s.a.i.AuditLogImpl ] [ec2-1-1-1-1] Message routing enabled: true

```

**Review by przemyslavic — APPROVED** [review]

**Review by erzetpe — APPROVED** [review]

**Pull request merged**